src/topology/tot_disc_eq_tot_sep.lean

```diff
@@ -0,0 +1,262 @@
+begin
+  set C := {Z : set α | is_clopen Z} with hC,
+  have h_open : ∀ Z ∈ C, is_open Z,
+  { rintros Z h, change (is_clopen Z) at h, apply h.1, },
+  have h_nhds : ∀(a:α) (U : set α), a ∈ U → is_open U → ∃Z ∈ C, a ∈ Z ∧ Z ⊆ U,
+  { rintros a U memU h_open,
+    obtain ⟨s, comp, xs, sU⟩ := exists_compact_subset h_open memU,
+    obtain ⟨t, h, ht, xt⟩ := mem_interior.1 xs,
+    set u : set s := (coe : s → α)⁻¹' (interior s) with hu,
+    have u_open_in_s : is_open u,
+    { apply is_open.preimage (continuous_subtype_coe) is_open_interior, },
+    obtain ⟨V, clopen_in_s, Vx, V_sub⟩ := @compact_space.compact_exists_clopen_in_open s _
+      (compact_iff_compact_space.1 comp) (subtype.t2_space) (subtype.totally_disconnected_space)
+        ⟨a, h xt⟩ u u_open_in_s xs,
+    have V_clopen : (set.image (coe : s → α) V) ∈ C,
+    { show is_clopen (set.image (coe : s → α) V), split,
+      { set v : set u := (coe : u → s)⁻¹' V with hv,
+        have : (coe : u → α) = (coe : s → α) ∘ (coe : u → s), { exact rfl, },
+        have pre_f : embedding (coe : u → α),
+        { rw this, refine embedding.comp embedding_subtype_coe embedding_subtype_coe, },
+        have f' : open_embedding (coe : u → α),
+        { constructor, apply pre_f,
+          { have : set.range (coe : u → α) = interior s,
+            { rw [this, set.range_comp],
+              have g : set.range (coe : u → s) = u,
+              { ext, split,
+                { rw set.mem_range, rintros ⟨y, hy⟩, rw ←hy, apply y.property, },
+                rintros hx, rw set.mem_range, use ⟨x, hx⟩, simp, },
+              rw [g, subtype.image_preimage_coe],
+              apply set.inter_eq_self_of_subset_left interior_subset, },
+            rw this, apply is_open_interior, }, },
+        have f2 : is_open v,
+        { rw hv, apply is_open.preimage (continuous_subtype_coe) clopen_in_s.1, },
+        have f3 : (set.image (coe : s → α) V) = (set.image (coe : u → α) v),
+        { rw [this, hv], symmetry',
+          rw set.image_comp, congr, rw [subtype.image_preimage_coe],
+          apply set.inter_eq_self_of_subset_left V_sub, },
+        rw f3, apply (open_embedding.is_open_map f') v f2, },
+      { apply (closed_embedding.closed_iff_image_closed
+          (is_closed.closed_embedding_subtype_coe (is_compact.is_closed comp))).1
+            (clopen_in_s).2, }, },
+    refine ⟨_, V_clopen, _, _⟩,
+    { simp [Vx, h xt], },
+    { transitivity s,
+      { simp, },
+      assumption, }, },
+  have f := topological_space.is_topological_basis_of_open_of_nhds h_open h_nhds,
+  use C, simp [f],
+end
+
+/-- A locally compact Hausdorff space is totally disconnected
+  if and only if it is totally separated. -/
+theorem loc_compact_t2_tot_disc_iff_tot_sep [locally_compact_space H] [t2_space H] :
+  totally_disconnected_space H ↔ totally_separated_space H :=
+begin
+  split,
+  { rintros h,
+    apply t2_space.tot_sep_of_zero_dim (@loc_compact_Haus_tot_disc_of_zero_dim _ _ _ _ h), },
```

Comment:
```suggestion
  { introI h,
    exact t2_space.tot_sep_of_zero_dim (loc_compact_Haus_tot_disc_of_zero_dim H) },
```
The `introI` tactic will put `h` is the type class system, see https://leanprover-community.github.io/mathlib_docs/tactics.html#Instance%20cache%20tactics if needed.

Reply:
Hello, apologies for the late reply (I was unwell). I have changed the code - shifted it into 2 files, instead of making a new file. Thank you very much for your suggestions, I have incorporated them, as well as theorems from the PR you mentioned above. This helped in cutting down the code by about 100 lines at least.

Reply:
@laughinggas, could you please mark all the resolved suggestions as resolved? If you're not sure about something, just reply.

Reply:
Apologies for that, it is now done.